src/main/java/com/o19s/es/ltr/query/RankerQuery.java

```diff
@@ -201,30 +206,32 @@ public void normalize(float norm, float boost) {
 
         @Override
         public RankerScorer scorer(LeafReaderContext context) throws IOException {
-            RankerChildScorer[] scorers = new RankerChildScorer[weights.length];
-            DocIdSetIterator[] subIterators = new DocIdSetIterator[weights.length];
-            for(int i = 0; i < weights.length; i++) {
+            ArrayList<Scorer> scorers = new ArrayList<>(weights.length);
+            List<DocIdSetIterator> subIterators = new ArrayList<>(weights.length);
+            for (int i = 0; i < weights.length; i++) {
                 Scorer scorer = weights[i].scorer(context);
                 if (scorer == null) {
                     scorer = new NoopScorer(this, context.reader().maxDoc());
                 }
-                scorers[i] = new RankerChildScorer(scorer, features[i]);
-                subIterators[i] = scorer.iterator();
+                scorers.add(scorer);
+                subIterators.add(scorer.iterator());
             }
             DocIdSetIterator rankerIterator = new NaiveDisjunctionDISI(DocIdSetIterator.all(context.reader().maxDoc()), subIterators);
             return new RankerScorer(scorers, rankerIterator);
         }
 
         class RankerScorer extends Scorer {
-            private final List<ChildScorer> scorers;
+            /**
+             * NOTE: Switch to ChildScorer and {@link #getChildren()} if it appears
+             * to be useful for logging
+             */
+            private final ArrayList<Scorer> scorers;
             private final DocIdSetIterator iterator;
-            private final float[] scores;
             private final LtrRanker.DataPoint dataPoint;
 
-            RankerScorer(RankerChildScorer[] scorers, DocIdSetIterator iterator) {
+            RankerScorer(ArrayList<Scorer> scorers, DocIdSetIterator iterator) {
```

Comment:
Yes... that's the problem, it hurts my eyes as well to see a var defined as `ArrayList<Type> var = ...`
But yes RandomAccess is critical here, and not sure what to choose between:
`Scorer[]` vs `ArrayList<Scorer>`
In the end what's important is random access and the fact that the array that holds the data is created once (not auto adjusted when building with things like ArrayList#ensureCapacity)